pkg/controller/selector/filter.go

```diff
@@ -57,20 +57,23 @@ func GetPoolCapabilityMap(pool *model.StoragePoolSpec) (map[string]interface{},
 		return nil, err
 	}
 
-	lastSimpleMap := make(map[string]interface{})
+	result := make(map[string]interface{})
 
 	for {
```

Comment:
Normally we don't use unlimited loop because it could cause the thread be stuck in it.